### PR TITLE
[iOS] Fix Bounds for PageContainer 

### LINKED ISSUE #6334: [Bug] iOS effect no longer works after upgrading to XF 4.0 since UIView.Bounds has no size in PlatformEffect.OnAttached()
### Description
I created an effect to add a gradient background to a UI element that no longer works in iOS after upgrading to XF 4.0 (the effect continued to work on Android). The cause of the issue is that the `UIView.Bounds` property started returning a `CGRect` with no size in the `PlatformEffect.OnAttached()` method.

I am working around this issue by rolling back to XF 3.6.

### Steps to Reproduce

1. Create an iOS `PlatformEffect` and apply it to a `ContentPage`.
2. Set breakpoint in `OnAttached()` method.

### Expected Behavior
`Container.Bounds` should have the same size as the `ContentPage`.

### Actual Behavior
`Container.Bounds` has no size (i.e. height and width are both 0).

### Basic Information

- Version with issue: 4.0.0.425677
- Last known good version: 3.6.0.344457
- IDE: VisualStudio

### LINKED ISSUE #6372: [Bug] [CustomRenderer] [iOS] View.Bounds all 0 OnElementChanged
### Description
Custom Renderer for a grandient background stopped working when upgrading to Xamarin Forms 4.x.x (from 3.6.x). View.Bounds width, height all 0

### Steps to Reproduce

1. Create custom iOS renderer deriving from PageRenderer class
2. Add override to OnElementChanged
3. Add code to render a grandient background

```
base.OnElementChanged(e);

            if (e.OldElement == null) // perform initial setup
            {
                var page = e.NewElement as GradientContentPage;

                var gradientLayer = new CAGradientLayer
                {
                    Frame = View.Bounds,
                    Colors = new CGColor[] { page.StartColor.ToCGColor(), page.EndColor.ToCGColor() }
                };

                View.Layer.InsertSublayer(gradientLayer, 0);
            }

```
### Expected Behavior
View.Bounds hat the actual width and height of the view. Adding a new CAGradientLayer used View.Bounds and had the current view's size.

### Actual Behavior
View.Bounds are all 0 (width, height) and thus the new Layer is not displayed (means: it's width and height are both 0)

### Basic Information

- Version with issue: 4.0.0.425677
- Last known good version: 3.6.x
- IDE: Visual Studio for Mac 8.8.8 (build 2)
- Platform Target Frameworks: 
  - iOS:  12.2
- Nuget Packages: FFImageLoading
- Affected Devices: iPhone

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change ###

We introduced a wrapper PageContainer and doing so we override the default LoadView of our PageRenderer ViewController. 

The default method provided a UIView with a Frame and Bounds, while our PageContainer doesn't provide one when it's created. Since our effects are attached when we set the elements for reuse on renderers when the user access the Bounds of the container there are not set ,while in previous versions there were (even if wrong ) .
Since these are related with size even previous versions should be using not the attach only but also the OnElementPropertyChanged or other methods to update there frame if the size changes. 

After some testing seems that the default size that was provided was the UIScreen even if inside a NavigationPage, this also is not very important as the Frame will be updated later on chain ViewWillLayoutSubviews.
### Issues Resolved ### 

- fixes #6334
- fixes #6372

### API Changes ###

 
 None

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

Should visit Issue 6334 and see a gradient and Success message.

### PR Checklist ###

- [ ] Has automated tests 
- [ ] Rebased on top of the target branch at time of PR
- [ ] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs [hunk 1]
```diff
@@ -123,8 +123,9 @@ public void SetElementSize(Size size)
 
 		public override void LoadView()
 		{
+			//by default use the MainScreen Bounds so Effects can access the Container size
 			if (_pageContainer == null)
-				_pageContainer = new PageContainer(this);
+				_pageContainer = new PageContainer(this) { Frame = UIScreen.MainScreen.Bounds};
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs]: I tested using the base.LoadView and see size created was the same as UIScreen.Bounds even inside a NavigationPage.

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6334.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+			Content = new Label
+			{
+				AutomationId = "IssuePageLabel",
+				Text = Fail
+			};
+		}
+
+#if UITEST && __IOS__
+		[Test]
+		public void Issue1Test()
```
Comment (samhouts) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6334.cs]:
```suggestion
		public void Issue334Test() 
```

Comment (rmarinho) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6334.cs]:
```suggestion
		public void Issue6334Test() 
```

File: Xamarin.Forms.Platform.iOS/ViewRenderer.cs [hunk 3]
```diff
@@ -40,7 +40,7 @@ public abstract class ViewRenderer<TView, TNativeView> : VisualElementRenderer<T
 		event EventHandler _controlChanging;
 		event EventHandler _controlChanged;
 
-
+		bool _isElementOrControlEmpty => (Element == null || Control == null);
```
Comment (paymicro) [inline Xamarin.Forms.Platform.iOS/ViewRenderer.cs]:
```suggestion
		bool _isElementOrControlEmpty => Element == null || Control == null;
```

Comment (rmarinho) [reply Xamarin.Forms.Platform.iOS/ViewRenderer.cs]: Sorry going to remove this to other PR so we can better track it

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6334.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+				AutomationId = "IssuePageLabel",
+				Text = Fail
+			};
+		}
+
+#if UITEST && __IOS__
+		[Test]
+		public void Issue6334Test() 
+		{
+			RunningApp.WaitForElement (q => q.Marked (AutomationId));
```
Comment (rmarinho) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6334.cs]:
```suggestion
			RunningApp.WaitForElement (q => q.Marked ("IssuePageLabel"));
```

**Pull request merged**